src/core/modules/players/players_entity.cpp

```diff
@@ -0,0 +1,702 @@
+/**
+* =============================================================================
+* Source Python
+* Copyright (C) 2012-2016 Source Python Development Team.  All rights reserved.
+* =============================================================================
+*
+* This program is free software; you can redistribute it and/or modify it under
+* the terms of the GNU General Public License, version 3.0, as published by the
+* Free Software Foundation.
+*
+* This program is distributed in the hope that it will be useful, but WITHOUT
+* ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+* FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+* details.
+*
+* You should have received a copy of the GNU General Public License along with
+* this program.  If not, see <http://www.gnu.org/licenses/>.
+*
+* As a special exception, the Source Python Team gives you permission
+* to link the code of this program (as well as its derivative works) to
+* "Half-Life 2," the "Source Engine," and any Game MODs that run on software
+* by the Valve Corporation.  You must obey the GNU General Public License in
+* all respects for all other code used.  Additionally, the Source.Python
+* Development Team grants this exception to all derivative works.
+*/
+
+// ============================================================================
+// >> INCLUDES
+// ============================================================================
+// Source.Python
+#include "players_entity.h"
+
+
+// ============================================================================
+// >> PlayerWrapper
+// ============================================================================
+boost::shared_ptr<PlayerWrapper> PlayerWrapper::__init__(unsigned int uiEntityIndex)
+{
+	CBaseEntityWrapper* pEntity = (CBaseEntityWrapper*) ExcBaseEntityFromIndex(uiEntityIndex);
+	if (!pEntity->IsPlayer())
+		BOOST_RAISE_EXCEPTION(PyExc_ValueError, "Index '%d' is not a valid player.");
+
+	return PlayerWrapper::wrap(pEntity->GetBaseEntity());
+}
+
+boost::shared_ptr<PlayerWrapper> PlayerWrapper::wrap(CBaseEntity* pEntity)
+{
+	return boost::shared_ptr<PlayerWrapper>(
+		(PlayerWrapper *) pEntity,
+		&NeverDeleteDeleter<PlayerWrapper *>
+	);
+}
+
+
+// CBasePlayer
+float PlayerWrapper::GetSpeed()
+{
+	static int offset = FindNetworkPropertyOffset("localdata.m_flLaggedMovementValue");
+	return GetNetworkPropertyByOffset<float>(offset);
+}
+
+void PlayerWrapper::SetSpeed(float value)
+{
+	static int offset = FindNetworkPropertyOffset("m_Local.m_bDucked");
```

Comment:
This should be `localdata.m_flLaggedMovementValue`.  However, when testing this branch with GunGame locally, it errored saying:

```python
[SP] Caught an Exception:
Traceback (most recent call last):
  File "..\addons\source-python\packages\source-python\events\listener.py", line 92, in fire_game_event
    callback(game_event)
  File "..\addons\source-python\plugins\gungame\plugins\included\gg_multi_level\gg_multi_level.py", line 156, in _player_level_up
    multi_level_manager.give_multi_level(player.userid)
  File "..\addons\source-python\plugins\gungame\plugins\included\gg_multi_level\gg_multi_level.py", line 128, in give_multi_level
    self[userid] = _MultiLevelPlayer.from_userid(userid)
  File "..\addons\source-python\packages\source-python\players\_base.py", line 99, in from_userid
    return cls(index_from_userid(userid))
  File "..\addons\source-python\plugins\gungame\plugins\included\gg_multi_level\gg_multi_level.py", line 59, in __init__
    self.speed = speed.get_int() / 100
  File "..\addons\source-python\packages\source-python\entities\_base.py", line 125, in __setattr__
    object.__setattr__(self, attr, value)

ValueError: Unable to find property 'm_Local.m_bDucked'.
```

So, I'm not sure if ducking is in a working state.  This was on CS:S.

Reply:
Good catch! I think it's still a valid property, but it's a datamap property, not a networked property. I will fix this when I get home on Friday.

Reply:
Fixed it via Github Web.